spec/models/course_spec.rb

```diff
@@ -167,5 +167,15 @@
         expect(subject.registration_key).to start_with('C')
       end
     end
+
+    describe '#users' do
+      let(:course) { create(:course, creator: owner, updater: owner) }
+      let(:un_approved_user) { create(:course_user, course: course).user }
+      let(:approved_user) { create(:course_user, :approved, course: course).user }
+      subject { course.users.with_approved_state }
```

Comment:
Shall I nest `#with_approved_state` under `#users` ?  Because these two functions are supposed to work together.